### ISSUE TITLE
Support no_std for Windows unwind information generation.

### ISSUE THREAD

**Issue by peterhuene** [CraneStation/cranelift#1203]
#### What is the feature or code improvement you would like to do in Cranelift?

With #1155 merged, the unwind information generation depends on `byteorder` and uses a trait that is explicitly gated by the `std` feature (`WriteBytesExt`).

The code in `isa/x86/unwind.rs` should be changed to not depend on this trait so that it the `byteorder` crate can be used with `no_std`.

#### What is the value of adding this in Cranelift?

Removes one more roadblock for `no_std` support.

#### Do you have an implementation plan, and/or ideas for data structures or algorithms to use?

Copy the tiny implementations of the convenience `write_xx` methods, as needed.

#### Have you considered alternative implementations? If so, how are they better or worse than your proposal

No alternatives considered.  To support `no_std`, we'll need to have the minimum implementation needed that doesn't depend on stdlib types.

**Comment by abrown** [comment]
Just thinking back to conversations I had with @sunfishcode re: vector endianness: perhaps I should have been using `byteorder` to properly write out the contents of vector constants, e.g.? There were a couple of places that endianness got baked in to the SIMD implemenation and perhaps we should open a separate issue for that?
